src/main/java/com/gooddata/warehouse/WarehouseService.java

```diff
@@ -546,27 +549,52 @@ public WarehouseS3Credentials getWarehouseS3Credentials(final Warehouse warehous
         notNull(s3Credentials, "s3Credentials");
         notNull(s3Credentials.getUri(), "s3Credentials.links.self");
 
-        final WarehouseTask task = createWarehouseTask(s3Credentials.getUri(), s3Credentials, HttpMethod.DELETE);
+        final WarehouseTask task = createDeleteCredentialsTask(s3Credentials.getUri(), s3Credentials);
         return new PollResult<>(this, createS3PollHandler(s3Credentials.getUri(), task, Void.class, "delete"));
     }
 
+    private WarehouseTask createAddCredentialsTask(final String targetUri,
```

Comment:
inline the method as it has only a single usage

Reply:
done